### PR TITLE
Introduce separate AMP & Jetpack integration classes

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

Introduce separate AMP & Jetpack integration classes to improve code quality.

## Relevant Technical Choices

As established by #5141, mostly moves some code around.

Based on #5141, so that will need to be merged first.

## To-do

N/A

## User-facing changes

N/A

## Testing Instructions

PHPUnit Tests

---

Related: #5062

### CODE REVIEW

File: includes/Plugin.php [hunk 1]
```diff
@@ -140,6 +142,20 @@ class Plugin {
...
+	 * @var AMP
+	 */
+	public $amp;
+
+	/**
+	 * Jetpack integration.
+	 *
+	 * @var Jetpack
+	 */
+	public $jetpack;
```
Comment (swissspidy) [inline includes/Plugin.php]: PHPMD rightfully complains that the `Plugin` class now has a lot of public fields.

Sounds like a refactor is due here.

Comment (spacedmonkey) [reply includes/Plugin.php]: Let's ignore for now and follow up in another ticket 👍

File: includes/Plugin.php [hunk 2]
```diff
@@ -140,6 +142,20 @@ class Plugin {
 	 */
 	public $experiments;
 
+	/**
+	 * AMP plugin integration.
+	 *
+	 * @var AMP
+	 */
+	public $amp;
```
Comment (spacedmonkey) [inline includes/Plugin.php]: I was thinking about this and having to many fields on a class. I how about idea. 

```
public $integrations = [];
```
then 
```
$this->integrations['amp'] = new AMP();
```

We would then break up this class into array of instances based on Namespace. integrations property for classes in the integrations namaspace etc. Then use a magic getting for backwards compatibility?

Comment (swissspidy) [reply includes/Plugin.php]: Yeah was basically thinking of the same; either 

```php
public $integrations = [];

$this->integrations['amp'] = new AMP();
```

or something like this

```php
public $integrations = new Integrations(); // or new stdObject

$this->integrations->amp = new AMP();
```

Everything beyond that (splitting up this class, magic getter, backwards compatibility, etc.) can be looked at in #5062

File: includes/Plugin.php [hunk 3]
```diff
@@ -218,6 +220,12 @@ public function register() {
...
 		$activation_notice->init();
 
+		$this->integrations['amp'] = new AMP();
+		add_action( 'init', [ $this->integrations['amp'], 'init' ] );
+
+		$this->integrations['jetpack'] = new Jetpack();
+		add_action( 'init', [ $this->integrations['jetpack'], 'init' ] );
+
 		$this->integrations['site-kit'] = new Site_Kit( $this->analytics );
 		add_action( 'init', [ $this->integrations['site-kit'], 'init' ] );
```
Comment (spacedmonkey) [inline includes/Plugin.php]: Can fix this lint error before I approve?

Comment (swissspidy) [reply includes/Plugin.php]: @spacedmonkey See discussion in #5141 about this.

This PR is WIP/DRAFT until #5141 is resolved

**Review by spacedmonkey — DISMISSED** [review]
PHPstan lint needs to be fixed.

File: includes/Story_Post_Type.php [hunk 4]
```diff
@@ -44,6 +44,7 @@
  *
```
Comment (spacedmonkey) [inline includes/Story_Post_Type.php:43]: I think we can remove these 
```
use DOMElement;
```

File: includes/Integrations/AMP.php [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+		}
+
+		$post = get_queried_object();
+		if ( ! ( $post instanceof WP_Post ) ) {
+			return $sanitizers;
+		}
+
+		$story = new Story();
+		$story->load_from_post( $post );
+		$sanitizers[ AMP_Story_Sanitizer::class ] = [
```
Comment (spacedmonkey) [inline includes/Integrations/AMP.php:128]: This lint also seem valid.

File: includes/Plugin.php (hunk 3 above)
Comment (spacedmonkey) [reply includes/Plugin.php]: This should fix it https://github.com/google/web-stories-wp/pull/5141/commits/aaebcdce349d24404b692cf212789c3fd0427a24

**Pull request merged**